### ISSUE TITLE
MySQL Server Fingerprinter

### ISSUE THREAD

**Issue by shreyamalviya** [guardicore/monkey#853]
Started with MySQL for #829.

In an attempt to find out the SSL requirements for a MySQL server by trying to authenticate, the following cases and results were obtained:
1. _SSL not configured on server_
    1.1 Right password, SSL used
           -------> `ERROR 2026 (HY000): SSL connection error: SSL is required but the server doesn't support it`
    1.2 Right password, SSL not used
           -------> :white_check_mark:
    1.3 Wrong password, SSL used
           -------> `ERROR 2026 (HY000): SSL connection error: SSL is required but the server doesn't support it`
    1.4 Wrong password, SSL not used
           -------> `ERROR 1045 (28000): Access denied for user '<username>'@'localhost' (using password: YES)`
2. _SSL configured on server, but not required for that user to connect to the server_
    2.1 Right password, SSL used
           -------> :white_check_mark:
    2.2 Right password, SSL not used
           -------> :white_check_mark:
    2.3 Wrong password, SSL used
           -------> `ERROR 1045 (28000): Access denied for user '<username>'@'localhost' (using password: YES)`
    2.4 Wrong password, SSL not used
           -------> `ERROR 1045 (28000): Access denied for user '<username>'@'localhost' (using password: YES)`
3. _SSL configured on server, and required for that user to connect to the server_
    3.1 Right password, SSL used
           -------> :white_check_mark:
    3.2 Right password, SSL not used
           ------->`ERROR 1045 (28000): Access denied for user '<username>'@'localhost' (using password: YES)`
    3.3 Wrong password, SSL used
           -------> `ERROR 1045 (28000): Access denied for user '<username>'@'localhost' (using password: YES)`
    3.4 Wrong password, SSL not used
           -------> `ERROR 1045 (28000): Access denied for user '<username>'@'localhost' (using password: YES)`

To summarize, it is possible for us to know if the server does not have SSL configured, but we can't know if the server _requires_ it, since SSL requirement is a user-specific thing in MySQL. It is considered a "privilege" that is granted to the user (see [this](https://dev.mysql.com/doc/refman/5.7/en/grant-tables.html#:~:text=ssl_type)).

To determine if SSL is disabled/not configured on the server, we can make use of cases 1.3, 2.3, and 3.3 from above.
